### PR TITLE
Add gRPC trimming end-to-end test

### PR DESCRIPTION
**Pull request by JamesNK**
This PR:

1. Updates gRPC interop tests to publish and then run tests using published output
2. Adds basic gRPC app that's published with trimming and then executed. Right now the trim warnings don't fail the build

### CODE REVIEW

File: src/Grpc/Linking/test/testassets/BasicWebsite/Program.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    public static IHostBuilder CreateHostBuilder(string[] args) =>
+        Host.CreateDefaultBuilder(args)
+            .ConfigureLogging(builder =>
+            {
+                builder.AddConsole(loggerOptions =>
+                {
+#pragma warning disable CS0618 // Type or member is obsolete
+                    loggerOptions.DisableColors = true;
+#pragma warning restore CS0618 // Type or member is obsolete
+                });
```
Comment (halter73) [inline src/Grpc/Linking/test/testassets/BasicWebsite/Program.cs:25]:
```suggestion
                builder.AddSimpleConsole(loggerOptions  =>
                {
                    loggerOptions.ColorBehavior = LoggerColorBehavior.Disabled;
                });
```

File: src/Grpc/Shared/ClientProcess.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        }
+    }
+
+    public void Dispose()
+    {
+        _processEx.Dispose();
+    }
+}
+
+public static class Publisher
```
Comment (halter73) [inline src/Grpc/Shared/ClientProcess.cs:91]: Why is this in ClientProcess.cs? It seems arbitrary WebsiteProcess isn't also in the same file.

**Review by halter73 — APPROVED** [review]

**Comment by JamesNK** [comment]
@dotnet/aspnet-build I'm modifying a test so that it does a `dotnet publish` and I get this error:

> 2022-07-07T15:47:32.2520949Z  ==> dotnet publish /datadisks/disk1/work/A39408F7/w/A4AE093B/e\testassets\InteropWebsite\InteropWebsite.csproj -r linux-x64 -c Release -o /datadisks/disk1/work/A39408F7/t/InteropWebsite --self-contained [/datadisks/disk1/work/A39408F7/w/A4AE093B/e]
> 2022-07-07T15:47:32.2521029Z  MSBuild version 17.3.0-preview-22322-07+334835f64 for .NET
> 2022-07-07T15:47:32.2521136Z    Determining projects to restore...
> 2022-07-07T15:47:32.2521186Z    All projects are up-to-date for restore.
> 2022-07-07T15:47:32.2521233Z  /datadisks/disk1/work/A39408F7/p/dotnet-cli/sdk/7.0.100-preview.7.22327.3/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/datadisks/disk1/work/A39408F7/w/A4AE093B/e/testassets/InteropWebsite/InteropWebsite.csproj]
> 2022-07-07T15:47:32.2521497Z  /datadisks/disk1/work/A39408F7/p/dotnet-cli/sdk/7.0.100-preview.7.22327.3/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(440,5): error NETSDK1082: There was no runtime pack for Microsoft.AspNetCore.App available for the specified RuntimeIdentifier 'linux-x64'. [/datadisks/disk1/work/A39408F7/w/A4AE093B/e/testassets/InteropWebsite/InteropWebsite.csproj]
> 2022-07-07T15:47:32.2521581Z  Process exited.

Problem:
> NETSDK1082: There was no runtime pack for Microsoft.AspNetCore.App available for the specified RuntimeIdentifier 'linux-x64'.

I thought this project had the shared FX built from the repo should present by the time it runs:

```xml
<!-- Enforce build order. Need shared Fx before PublishTestAssets runs. -->
<ProjectReference Include="$(RepoRoot)\src\Framework\App.Ref\src\Microsoft.AspNetCore.App.Ref.csproj"
  Private="false"
  ReferenceOutputAssembly="false"
  SkipGetTargetFrameworkProperties="true" />
```

Is there a way I can get `dotnet publish` to work?

**Comment by dougbu** [comment]
> Is there a way I can get `dotnet publish` to work?

The short answer is "yes, but&hellip;`. What's happening here is likely problems attempting to use a project in a different directory than the work item root. Only projects directly below that will pick up the various hacks in the Directory.Build.* files we push to Helix. Those hacks are necessary to ensure the latest runtimes are used.

I believe we have a mix of solutions in the repo. Some tests use temporary projects w/in the work item folder while others copy files like global.json, NuGet.config, and Directory.Build.* into (or above) temporary projects located elsewhere.

**Pull request closed without merging**